src/core/server/config/env.ts

```diff
@@ -115,6 +115,7 @@ export class Env {
 
     this.pluginSearchPaths = [
       resolve(this.homeDir, 'src', 'plugins'),
+      resolve(this.homeDir, 'x-pack', 'plugins'),
```

Comment:
Hardcoding this into core will make it so it's impossible to develop on only the OSS version of Kibana, so we'll need to inform core of this option at instantiation time and configure it based on the circumstance. I'm going to throw up a dedicated PR for this.